tests/executor/exchange_hash_test.cpp

```diff
@@ -0,0 +1,422 @@
+    right_table_logical_tile_ptrs.push_back(
+        std::move(right_table_logical_tile));
+  }
+
+  // Right scan executor returns logical tiles from the right table
+
+  EXPECT_CALL(right_table_scan_executor, DInit()).WillOnce(Return(true));
+
+  ExpectNormalTileResults(right_table_tile_group_count,
+                          &right_table_scan_executor,
+                          right_table_logical_tile_ptrs);
+
+  // Create hash keys
+  expression::AbstractExpression *right_table_attr_1 =
+      new expression::TupleValueExpression(1, 1);
+
+  std::vector<std::unique_ptr<const expression::AbstractExpression>> hash_keys;
+  hash_keys.emplace_back(right_table_attr_1);
+
+  // Create hash plan node
+  planner::HashPlan hash_plan_node(hash_keys);
+
+  // Construct the hash executor
+  executor::HashExecutor hash_executor(&hash_plan_node, nullptr);
+  hash_executor.AddChild(&right_table_scan_executor);
+
+  EXPECT_TRUE(hash_executor.Init());
+  EXPECT_TRUE(hash_executor.Execute());
+
+  // Create a same copy for the other hash executor
+  MockExecutor right_table_scan_executor2;
+
+  std::vector<std::unique_ptr<executor::LogicalTile>>
+      right_table_logical_tile_ptrs2;
+
+  for (size_t right_table_tile_group_itr = 0;
+       right_table_tile_group_itr < right_table_tile_group_count;
+       right_table_tile_group_itr++) {
+    std::unique_ptr<executor::LogicalTile> right_table_logical_tile(
+        executor::LogicalTileFactory::WrapTileGroup(
+            right_table->GetTileGroup(right_table_tile_group_itr)));
+    right_table_logical_tile_ptrs2.push_back(
+        std::move(right_table_logical_tile));
+  }
+
+  // Right scan executor returns logical tiles from the right table
+
+  EXPECT_CALL(right_table_scan_executor2, DInit()).WillOnce(Return(true));
+
+  ExpectNormalTileResults(right_table_tile_group_count,
+                          &right_table_scan_executor2,
+                          right_table_logical_tile_ptrs2);
+
+  // Create hash keys
+  expression::AbstractExpression *right_table_attr_12 =
+      new expression::TupleValueExpression(1, 1);
+
+  std::vector<std::unique_ptr<const expression::AbstractExpression>> hash_keys2;
+  hash_keys2.emplace_back(right_table_attr_12);
+
+  // Create hash plan node
+  planner::HashPlan hash_plan_node2(hash_keys2);
+
+  // Construct the hash executor
+  executor::ExchangeHashExecutor parallel_hash_executor(&hash_plan_node2,
+                                                        nullptr);
+  parallel_hash_executor.AddChild(&right_table_scan_executor2);
+
+  EXPECT_TRUE(parallel_hash_executor.Init());
+  EXPECT_TRUE(parallel_hash_executor.Execute());
+
+  // Compare the result hash table of the two hash executors
+  // Loop through table 1, ensure that every entry in table 1 is in table 2.
+  // Delete the corresponding entry in table 2 for every entry seen in table 1
+  // Ensure in the end we have an empty table 2
+  auto &hash_table = hash_executor.GetHashTable();
```

Comment:
This tests your code with the assumption that the sequential hash is correct, It might be a good idea to remove this dependency.